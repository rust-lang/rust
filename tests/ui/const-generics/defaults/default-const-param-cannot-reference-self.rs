struct Struct<const N: usize = { Self; 10 }>;
//~^ ERROR generic parameters cannot use `Self` in their defaults [E0735]

enum Enum<const N: usize = { Self; 10 }> { }
//~^ ERROR generic parameters cannot use `Self` in their defaults [E0735]

union Union<const N: usize = { Self; 10 }> { not_empty: () }
//~^ ERROR generic parameters cannot use `Self` in their defaults [E0735]

fn main() {
    let _: Struct;
    let _: Enum;
    let _: Union;
}
