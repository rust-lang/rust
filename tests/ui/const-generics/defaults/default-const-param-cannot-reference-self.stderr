error[E0735]: generic parameters cannot use `Self` in their defaults
  --> $DIR/default-const-param-cannot-reference-self.rs:1:34
   |
LL | struct Struct<const N: usize = { Self; 10 }>;
   |                                  ^^^^ `Self` in generic parameter default

error[E0735]: generic parameters cannot use `Self` in their defaults
  --> $DIR/default-const-param-cannot-reference-self.rs:4:30
   |
LL | enum Enum<const N: usize = { Self; 10 }> { }
   |                              ^^^^ `Self` in generic parameter default

error[E0735]: generic parameters cannot use `Self` in their defaults
  --> $DIR/default-const-param-cannot-reference-self.rs:7:32
   |
LL | union Union<const N: usize = { Self; 10 }> { not_empty: () }
   |                                ^^^^ `Self` in generic parameter default

error: aborting due to 3 previous errors

For more information about this error, try `rustc --explain E0735`.
